src/main/java/ch/bfh/bti7081/s2018/green/presenters/MedicationPrescriptionPresenter.java

```diff
@@ -15,43 +15,41 @@
 
     private DataContainer data;
     private MedicationPrescriptionView view;
-    public MedicationPrescriptionPresenter(MedicationPrescriptionView view) {
+    private MedicationPresenter viewBehind;
+    public MedicationPrescriptionPresenter(MedicationPrescriptionView view, MedicationPresenter viewBehind) {
         this.view = view;
-       
+        this.viewBehind = viewBehind;
         this.data = DataContainer.getInstance();
-        
+
         view.getSaveButton().addClickListener(click -> {
             this.saveData();
         });
-        
     }
-    
+
 
     private void saveData (){
-    	
-    	try {
-        Medication medication = view.getMedication();
-        medication.setPatient(data.getCurrentPatient());
-        medication.setPrescriber(data.getCurrentStaff());
-        MedicationManager manager = new MedicationManager();
-        if (medication.getId() == 0) {
-        	manager.add(medication);	
-        } else {
-        	manager.update(medication);
+        try {
+            Medication medication = view.getMedication();
+            if (medication != null)
+            {
+                medication.setPatient(data.getCurrentPatient());
+                medication.setPrescriber(data.getCurrentStaff());
+                MedicationManager manager = new MedicationManager();
+                if (medication.getId() == 0) {
+                    manager.add(medication);
+                } else {
+                    manager.update(medication);
+                }
+
+                // Show success-message in case saving was successful
+                Notification.show("Data was saved!");
+                viewBehind.showExpiredMedication();
```

Comment:
Why do you change user settings upon save (confuses the user)? If you want to notify the user if his saved medication is hidden, then check if the 'to' date isBefore now() and if the filter is activated and then show a notification. But since we don't have time before our deadline just leave user settings and don't notify.

Reply:
? Cannot follow your comment, initially most oft this code snipped was done by you if I'm not misstaken?